functions/index.js

```diff
@@ -152,26 +153,48 @@ exports.dialogflowFirebaseFulfillment = functions.https.onRequest((request, resp
 
     function welcomeMessage(agent) {
         agent.add(`Greetings! What Heartfulness event are you reporting on?
-        \nFor example, you can enter Dhyanotsav, U-Connect, AtWork, C-Connect, V-Connect, G-Connect, Kaushalam, CME, Youth, Yoga, Temple, Legal, Family, NGO, Brighter Minds, etc. 
-        \nFor general Heartfulness Introductory Events, enter "Heartfulness". 
-        \nFor *School or S-Connect* events, enter which program: HELP, INSPIRE, HEART or THWC
+        \nFor example, you can enter Dhyanotsav, U-Connect, AtWork, C-Connect, V-Connect, G-Connect, CME, Youth, Yoga, Temple, Legal, Family, NGO, Brighter Minds, etc. 
+        \nFor general Heartfulness Introductory Events, just enter "Heartfulness". 
+        \nFor School or S-Connect events, enter which program: HELP, INSPIRE, HEART or THWC
+        \nFor Group Meditations, simply enter "Satsangh" or "Group Meditation"
         \nIf you don't know just enter 'Other'.`);
         return;
 
     }
 
     function askForConfirmation(agent) {
         let type = agent.parameters['event_type'];
+        let event_day = agent.parameters['event_day'];
         let count = agent.parameters['event_count'];
         let name = agent.parameters['coordinator_name'];
+        let phone = agent.parameters['coordinator_phone'];
         let isoDateString = agent.parameters['event_date'];
         let institution = agent.parameters['event_institution'];
-        let center = findCity(agent.parameters['event_city']);
-        let date = cleanDate(isoDateString);
-        console.log(center.city, center.zone, center.country);
-        agent.add(`Okay, ${count} attended ${type} on ${date} at ${institution} in ${center.city}, Is this correct ${name}? Please reply with 'yes' or 'no'.`);
-        return;
+        let city = agent.parameters['event_city'];
+        let feedback = agent.parameters['event_feedback'];
+
+		if(type) {
+          	console.log(type, event_day);
+	      	if (type != "group-meditation" && !event_day){
+              	console.log("Need which day");
+				agent.add(`Is is day-1, day-2 or day-3 of the event? If it is a one day event just enter \"one day event\". For a follow up event, enter \"Follow Up\"`);			
+            }
+        }
+
+        let allParametersReady = false;
+      	// Check if we have all required params
+        if (type && count && name && phone && isoDateString && institution &&  city && feedback) {
```

Comment:
Instead of having a mutable variable, you can simply convert it to a function. 

```JS
const allParamsReady = () => {
    return type && count && name && phone && isoDateString && institution &&  city && feedback
}
```

and simply do :

```JS
if (allParamsReady()) {
   // do something
}
```